src/components/layout/RegisterNameForm/RegisterNameForm.tsx

```diff
@@ -0,0 +1,163 @@
+import { useState } from 'react';
+
+import { useRegistrationState } from '../../../state/contexts/RegistrationState';
+import {
+  isArNSDomainNameValid,
+  isArweaveTransactionID,
+} from '../../../utils/searchUtils';
+import Dropdown from '../../inputs/Dropdown/Dropdown';
+import SearchBar from '../../inputs/Search/SearchBar/SearchBar';
+import UpgradeTier from '../UpgradeTier/UpgradeTier';
+import './styles.css';
+
+function RegisterNameForm() {
+  const [antFlow, setAntFlow] = useState(
+    'Create an Arweave Name Token (ANT) for me',
+  );
+  const [antAddress, setAntAddress] = useState(0);
+  const [{ domain, ttl, targetID }, dispatchRegisterState] =
+    useRegistrationState();
+
+  function updateRegisterState({ key, value }: { key: any; value: any }) {
+    setTimeout(
+      () =>
+        dispatchRegisterState({
+          type: key,
+          payload: value,
+        }),
+      50,
+    );
+  }
+
+  function handleTargetID(id: string) {
+    // todo: add some more advanced error handling and notifications to check if id is a dataID or a value transfer
+    if (isArweaveTransactionID(id)) {
+      updateRegisterState({ key: 'setTargetID', value: id });
+    }
+    if (!id) {
+      updateRegisterState({ key: 'setTargetID', value: undefined });
+    }
+    if (!isArweaveTransactionID(id)) {
+      updateRegisterState({ key: 'setTargetID', value: undefined });
+    }
+  }
+
+  return (
+    <>
+      <div className="register-name-modal center">
+        <div className="section-header">{domain} is available!</div>
+
```

Comment:
nit: surprised the husky didn't catch these extra spaces 🤔

Reply:
human formatter activated, spaces destroyed